samples/kotlin-android-app-destinations/src/test/java/com/segment/analytics/destinations/plugins/IntercomDestinationTest.kt

```diff
@@ -0,0 +1,362 @@
+package com.segment.analytics.destinations.plugins
+
+import android.app.Application
+import android.app.TaskStackBuilder
+import android.util.Log
+import com.segment.analytics.kotlin.core.*
+import com.segment.analytics.kotlin.core.platform.Plugin
+import com.segment.analytics.kotlin.core.utilities.getString
+import io.intercom.android.sdk.Company
+import io.intercom.android.sdk.Intercom
+import io.intercom.android.sdk.UnreadConversationCountListener
+import io.intercom.android.sdk.UserAttributes
+import io.intercom.android.sdk.helpcenter.api.CollectionContentRequestCallback
+import io.intercom.android.sdk.helpcenter.api.CollectionRequestCallback
+import io.intercom.android.sdk.helpcenter.api.SearchRequestCallback
+import io.intercom.android.sdk.identity.Registration
+import io.mockk.*
+import io.mockk.impl.annotations.MockK
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.TestCoroutineScope
+import kotlinx.serialization.json.*
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertNotNull
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import java.io.ByteArrayInputStream
+import java.net.HttpURLConnection
+
+internal class IntercomDestinationTest {
+
+    @MockK(relaxUnitFun = true)
+    lateinit var application: Application
+
+    private lateinit var intercom: Intercom
+
+    private var configuration: Configuration
+
+    private lateinit var analytics: Analytics
+
+    private lateinit var intercomDestination: IntercomDestination
+
+    private val testDispatcher = TestCoroutineDispatcher()
+
+    private val testScope = TestCoroutineScope(testDispatcher)
+
+    init {
+        MockKAnnotations.init(this)
+
+        // mock intercom
+        mockkStatic(Intercom::class)
+        intercom = spyk<Intercom>(StubIntercom())
```

Comment:
I had tests failed because of direct mock, so I changed it to spy. I just tested mock with `relaxUnitFun = true` that works well. will remove the stub